src/components/Permissions/AppPermissions.jsx

```diff
@@ -96,8 +99,21 @@ const PermissionsApplication = ({ t }) => {
     matchingQueryResult = queryResultKonnectors
   }
 
-  const sortPermissionsByName = queryResult => {
-    return Object.entries(queryResult.data?.permissions)
+  let limitedRightAccess = []
+  let exitRights = []
+  if (remoteDoctypes && matchingQueryResult?.data?.permissions) {
+    const permissions = Object.entries(matchingQueryResult.data.permissions)
+    for (let i = 0; i < permissions.length; i++) {
+      if (remoteDoctypes.includes(permissions[i][1].type)) {
+        exitRights.push(permissions[i])
+      } else {
+        limitedRightAccess.push(permissions[i])
+      }
+    }
+  }
```

Comment:
@vdnj Est-ce que tu peux sortir ça du composant stp ? 

Aussi `exitRights` ça veut dire quoi ? Si c'est pour parler de sortir de la donnée, je parlerai plus de `get out` ou un truc du genre

Reply:
Tu veux que je sorte toute la for loop ? Dans une fonction que je mets dans un helper alors ?

Reply:
Yep. 

Je répète encore : Tout ce qui n'a pas besoin du cycle de vie de React (pas besoin des contextes, pas besoin des setState) doit être en dehors du composant React ! Ça évite pas mal de chose : 
- des redéclarations de fonction (sinon à chaque render ça re-crée encore et toujours les méthodes) 
- d'écrire des tests compliques. C'est beaucoup plus simple d'avoir une méthode tout simple avec des arguments et de tester la sortie que de testes de composants avec des hooks et tout 
- d'avoir des composants illisibles 
- ça permet de faciliter la réutilisation.